apps/roam/src/utils/cleanupOrphanedNodes.ts

```diff
@@ -1,39 +1,29 @@
-import { getSupabaseContext } from "./supabaseContext";
-import { getNodeEnv } from "roamjs-components/util/env";
+import { type SupabaseClient } from "@supabase/supabase-js";
+import { getSupabaseContext, getLoggedInClient } from "./supabaseContext";
 
-const getAllNodesFromSupabase = async (): Promise<string[]> => {
+const getAllNodesFromSupabase = async (
+  spaceId: number,
+  supabaseClient: SupabaseClient,
+): Promise<string[]> => {
   try {
-    const context = await getSupabaseContext();
-    if (!context) {
-      console.error("Failed to get Supabase context");
-      return [];
-    }
-
-    const baseUrl =
-      getNodeEnv() === "development"
-        ? "http://localhost:3000/api/supabase"
-        : "https://discoursegraphs.com/api/supabase";
+    const allNodeInstanceIds = await supabaseClient
+      .from("Concept")
+      .select("source_local_id")
+      .not("schema_id", "is", null);
```

Comment:
Puzzled; this constraint means that you're looking at node instances, and not schemas. At line 16 you mention node schemas, I suspect that's just a typo. But can't schemas also be orphaned? Why restrict to instances?

Reply:
Yeah for schemas I would have a separate pr, this task was pending for weeks and was originally only for node instances. I will have the schema  pr this week